v2/pkg/engine/datasource/graphql_datasource/graphql_sse_handler.go

```diff
@@ -52,11 +53,24 @@ func (h *gqlSSEConnectionHandler) StartBlocking(sub Subscription) {
 
 	go h.subscribe(reqCtx, sub, dataCh, errCh)
 
+	var ticker *time.Ticker
+	if sub.options.SendHeartbeat {
+		ticker = time.NewTicker(resolve.HearbeatInterval)
+	}
+
 	for {
 		select {
+		case <-tickerC(ticker):
+			sub.updater.Heartbeat()
 		case data := <-dataCh:
+			if ticker != nil {
```

Comment:
It would be great to document the current design. This allows to avoid sending heartbeats when the subgraph has already send data.

Reply:
I added an additional comment in the Heartbeat godoc. Is that what you had intended?